spec/mempool/gossip/flood.qnt

```diff
@@ -0,0 +1,143 @@
+// -*- mode: Bluespec; -*-
+
+/*******************************************************************************
+Flood is a basic _push_ gossip protocol: every time a node receives a
+transaction, it forwards (or "pushes") the transaction to all its peers, except
+to the peer(s) from which it received the transaction.
+*******************************************************************************/
+module flood {
+    import spells.* from "./spells"
+    import mempool.* from "./mempool"
+    export mempool.*
+
+    //--------------------------------------------------------------------------
+    // Messages
+    //--------------------------------------------------------------------------
+
+    // Nodes communicates only one type of message carrying a full transaction.
+    type Message =
+        | TxMsg(TX)
+
+    //--------------------------------------------------------------------------
+    // Actions
+    //--------------------------------------------------------------------------
+
+    // Upon receiving a transaction message from a peer, the node attempts to
+    // add the transaction to its mempool.
+    action handleMessage(node, _msgs, sender, msg) =
```

Comment:
So `_msgs` here is of the same time of `var msgs` in `p2p.qnt`?

Reply:
`_msgs` has the same type as `msgs` in the `p2p` module. This parameter is to pass updated versions of `msgs`, so first `receiveFromPeer` removes the first message in the queue and pass the tail to `handleMessage`, later in dog.qnt, `_msgs` may updated with a `HaveTx` message. I clarified in the text.